plugins/jira-dashboard-backend/src/api.ts

```diff
@@ -93,26 +96,53 @@ export type SearchOptions = {
  * @param config - A Backstage config
  * @param jqlQuery - A string containing the jql query.
  * @param options - Query options that will be passed on to the POST request.
- *
+ * @param logger - An optional logger for error handling and debugging purposes.
+ * @returns A promise that resolves with the search results and status code.
+ * @throws If an error occurs during the search process.
  * @public
  */
 export const searchJira = async (
   config: Config,
   jqlQuery: string,
   options: SearchOptions,
-): Promise<Issue[]> => {
-  const response = await fetch(`${resolveJiraBaseUrl(config)}search`, {
-    method: 'POST',
-    body: JSON.stringify({ jql: jqlQuery, ...options }),
-    headers: {
-      Authorization: resolveJiraToken(config),
-      Accept: 'application/json',
-      'Content-Type': 'application/json',
-    },
-  }).then(resp => resp.json());
-  return response.issues;
+  logger: Logger,
+): Promise<SearchJiraResponse> => {
+  try {
+    const response = await fetch(`${resolveJiraBaseUrl(config)}search`, {
+      method: 'POST',
+      body: JSON.stringify({ jql: jqlQuery, ...options }),
+      headers: {
+        Authorization: resolveJiraToken(config),
+        Accept: 'application/json',
+        'Content-Type': 'application/json',
+      },
+    });
+    const jsonResponse = await response.json();
+
+    return {
+      results: toPaginatedResponse(jsonResponse),
```

Comment:
Can we just return the "jsonResponse" directly?

Reply:
Resolved :)